packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -34,6 +34,61 @@ import {
 } from './shared/shared-hook-functions';
 import { toSaveSettings } from './to-save-settings';
 
+function mergeHookFunctions(...hookFunctions: IWorkflowExecuteHooks[]): IWorkflowExecuteHooks {
+	const result: IWorkflowExecuteHooks = {
+		nodeExecuteBefore: [],
+		nodeExecuteAfter: [],
+		workflowExecuteBefore: [],
+		workflowExecuteAfter: [],
+		sendResponse: [],
+		nodeFetchedData: [],
+	};
+	for (const hooks of hookFunctions) {
+		for (const key in hooks) {
+			if (hooks.hasOwnProperty(key)) {
+				result[key]!.push(...hooks[key]!);
+			}
+		}
+	}
+	return result;
+}
+
+function hookFunctionsWorkflowTelemetry(userId?: string): IWorkflowExecuteHooks {
+	const eventService = Container.get(EventService);
+	return {
+		workflowExecuteBefore: [
+			async function (this: WorkflowHooks): Promise<void> {
+				const { executionId, workflowData } = this;
+				eventService.emit('workflow-pre-execute', { executionId, data: workflowData });
+			},
+		],
+		workflowExecuteAfter: [
+			async function (this: WorkflowHooks, runData: IRun): Promise<void> {
+				const { executionId, workflowData: workflow } = this;
+				eventService.emit('workflow-post-execute', { executionId, runData, workflow, userId });
+			},
+		],
+	};
+}
+
+function hookFunctionsNodeTelemetry(): IWorkflowExecuteHooks {
```

Comment:
`telemetry` is inaccurate because more things happen in reaction to these events, e.g. log streaming. `hookFunctionsNodeEvents`?

Reply:
Same applies to the other functions here as well then right? The producer of the events shouldn't be aware of what the consumers do, so tying the naming of these to what happens on the consumer side can be confusing. So I agree with Ivan, but maybe lets extend it to all the hook creating functions here

Reply:
I think we've been over this - the naming in lifecycle hooks is maddening, but I'd say renaming all bundles is out of scope for this PR. I focused on `hookFunctionsNodeTelemetry` because it's being introduced here.